app/src/main/java/com/example/atheneum/fragments/SearchFragment.java

```diff
@@ -82,80 +56,41 @@
-                String username = ((TextView) view.findViewById(android.R.id.text1)).getText().toString();
-                Log.d(TAG, username + " was selected");
-
-                db = FirebaseDatabase.getInstance();
-                dbRef = db.getReference("users");
-                dbRef.orderByChild("userName").equalTo(username).addListenerForSingleValueEvent(new ValueEventListener() {
-                    @Override
-                    public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-                        for (DataSnapshot child: dataSnapshot.getChildren()) {
-                            User selectedUser = child.getValue(User.class);
-                            ((MainActivity)getActivity()).passDataToViewProfileActivity(selectedUser);
-                        }
-                    }
-
-                    @Override
-                    public void onCancelled(@NonNull DatabaseError databaseError) {
-
-                    }
-                });
-            }
-        });
-
-        userList = new ArrayList<>();
-        final ArrayList<String> userNameList = new ArrayList<>();
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
+        linearLayoutManager.setOrientation(LinearLayoutManager.VERTICAL);
 
-        dbRef.addValueEventListener(new ValueEventListener() {
+        userListRecyclerView = view.findViewById(R.id.user_recyclerview);
+        userListRecyclerView.setLayoutManager(linearLayoutManager);
+        userListRecyclerView.addItemDecoration(new DividerItemDecoration(userListRecyclerView.getContext(), DividerItemDecoration.VERTICAL));
+        userListAdapter = new UserListAdapter();
+        userListRecyclerView.setAdapter(userListAdapter);
+        userListAdapter.setOnClickListener(new UserListAdapter.onClickListener() {
             @Override
-            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-                Log.d(TAG, "On Data Change was Called");
-                for (DataSnapshot child : dataSnapshot.getChildren()) {
-                    User aUser = child.getValue(User.class);
-                    userList.add(aUser);
-                    userNameList.add(aUser.getUserName());
+            public void onClick(@NonNull User user) {
+                if (getActivity() instanceof MainActivity) {
+                    ((MainActivity)getActivity()).passDataToViewProfileActivity(user);
                 }
-                defaultUserNameList = userNameList;
-                ArrayAdapter<String> adapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, userNameList);
-                userListView.setAdapter(adapter);
             }
+        });
 
+        searchUsersViewModel = ViewModelProviders.of(getActivity()).get(SearchUsersViewModel.class);
+        LiveData<List<User>> userListLiveData = searchUsersViewModel.getUserListLiveData();
```

Comment:
I was mentioning this in my previous comment. The google repo (https://github.com/googlesamples/android-architecture-components/blob/master/BasicSample/app/src/main/java/com/example/android/persistence/ui/ProductFragment.java) for their UI on lines 85 and 93 show them calling a method to get the LiveData and calling the .observe() method immediately. I tested removing this and simply replacing line 83 and 84 with the code below and it seemed to work fine.
```
searchUsersViewModel.getUserListLiveData().observe(...);
```

Reply:
Fixed